crate2nix/Cargo.nix

```
@@ -7,13 +7,14 @@
 , pkgs ? import nixpkgs { config = {}; }
 , lib ? pkgs.lib
 , stdenv ? pkgs.stdenv
-, buildRustCrate ? pkgs.buildRustCrate
```

Is it necessary to remove this argument? If yes, why?

[I just like to keep top-level interface changes minimally, and for whatever reason I made this a top-level argument.]

---

@kolloch see what @lopsided98 said in the PR description:

> Unfortunately, I haven't figured out a way to do this without potentially breaking existing users, as well as the debugging tools. This is because there is no longer a single version of `buildRustCrate` used by all crates, and instead there are separate ones for each platform. This makes it difficult to pass around `buildRustCrateFunc`.

I can attest that this makes sense from the usual cross perspective.

---

We can do
```
buildRustCrate' ? pkgs: pkgs.buildRustCrate
```
if you prefer, @kolloch. That will work.

---

and `buildRustCrate` could be deprecated and given a `null` default, so we can migrate with
```suggestion
, buildRustCrate ? null
, buildRustCrateFromPkgs ? if buildRustCrate != null then _: buildRustCrate else traceDeprecatedMessage (pkgs: pkgs.buildRustCrate)
```
With `traceDeprecatedMessage` left as a bikeshed to the reader :).